bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleEngine.java

```diff
@@ -328,18 +328,24 @@ protected synchronized void setRule(String rUID) {
             errMsgs = errMsgs + "\n" + errMessage;
         }
 
-        try {
-            ConnectionValidator.validateConnections(r);
-        } catch (Exception e) {
-            errMsgs = errMsgs + "\n Validation of rule" + r.getUID() + "has failed! " + e.getMessage();
+        if (errMsgs == null) {
+            try {
+                ConnectionValidator.validateConnections(r);
+            } catch (IllegalArgumentException e) {
+                unregister(r);
```

Comment:
It is required, because unregister method removes handlers associated to the modules